src/osd/SnapMapper.cc

```diff
@@ -658,11 +659,10 @@ vector<hobject_t> SnapMapper::get_objects_by_prefixes(
 
       ceph_assert(is_mapping(next.first));
 
-      if (auto next_prefix = next.first.substr(0, prefix.size());
-          next_prefix != prefix) {
+      if (next.first.compare(0, prefix.size(), prefix) != 0) {
 	// TBD: we access the DB twice for the first object of each iterator...
 	dout(20) << fmt::format("{}: breaking, prefix expected {} got {}",
-	                        __func__, prefix, next_prefix)
+	                        __func__, prefix, next.first)
```

Comment:
nit: This log line is slightly not accurate with the changes here.
"breaking, prefix expected {} got key {} with a different prefix"